cmake_file_api/kinds/cache/v2.py

```diff
@@ -38,17 +38,17 @@ def __repr__(self) -> str:
         )
 
 
-class CacheEntry(object):
+class CacheEntry:
     __slots__ = ("name", "value", "type", "properties")
 
-    def __init__(self, name: str, value: str, type: CacheEntryType, properties: List[CacheEntryProperty]):
+    def __init__(self, name: str, value: str, type: CacheEntryType, properties: list[CacheEntryProperty]):
         self.name = name
         self.value = value
         self.type = type
         self.properties = properties
 
     @classmethod
-    def from_dict(cls, dikt: Dict) -> "CacheEntry":
+    def from_dict(cls, dikt: dict) -> "CacheEntry":
```

Comment:
Also, does the `dikt` need a more specialized typing annotation?
e.g. `dict[str, object]`?

I see you've changed it into `dict[str, Any]` in other locations.

Reply:
Yes, needs to be changed everywhere to comply. Long-term it would be great to replace the Any as well, perhaps

Reply:
When/if that becomes possible, you'd effectively replace a lot of work these classes do.